### PR TITLE
Add integration interactive test for debugging intermittent test failures

### LINKED ISSUE #138: [BUG] Integration Test Non-Repoducible Behaviour
### Short summary

Many Integration Tests will Fail Randomly in ways that are difficult to track

### Steps to reproduce

- Antenna Deployer Test 5
- Power Monitor Test 1 and 2
- RTC Test 3
- IMU test 2

### Expected behavior

Could be related to Hardware Use
Could also be related to cube setup -- what files exist and such
Has been Race Conditions in many cases we've caught so could still be that as well
Could be edge cases we really need to figure out!

### Actual behavior

j

### Logs / crash output

_No response_

### Target board

RP2350 (Pico 2)

### PR DESCRIPTION
**Pull request by yudataguy**
## Summary

Adds a interactive integration test runner to help debug non-reproducible test failures reported in #138.

## Problem

Several integration tests fail randomly in ways that are difficult to track:
- Antenna Deployer Test 5
- Power Monitor Tests 1 and 2
- RTC Test 3
- IMU Test 2

These failures could be due to race conditions, hardware timing issues, or edge cases.
These tests have been fixed in #321 For future test, this interactive integration test can help debug any potential flaky tests during multiple runs. 

## Solution

This PR introduces a interactive test runner that executes tests multiple times to detect and analyze flaky behavior.

### Features

  Add interactive integration test runner with auto-discovery and flaky test detection                                                     
                                                                                                                                           
  Replaces hardcoded test approach with flexible test runner that:                                                                         
  - Auto-discovers all integration tests from test/int directory                                                                           
  - Provides interactive cursor-based selection (arrow keys + spacebar)                                                                    
  - Supports CLI mode for automation and scripts                                                                                           
  - Runs tests for configurable cycles to detect flaky behavior                                                                            
  - Automatically identifies and reports flaky tests                                                                                       
  - Shows detailed statistics including pass rates and timing                                                                              
  - Gracefully handles errors with clear messages  

### Usage

  - Interactive: make test-interactive (default cycle: 3)                                                                                                     
  - CLI: make test-interactive ARGS="--tests watchdog_test --cycles 10"                                                                    
  - All tests: make test-interactive ARGS="--all --cycles 20"      

### Output Example

Screenshot 2026-02-03 at 3 40 59 PM

```
================================================================================
TEST SUMMARY
================================================================================

Test: antenna_deployer_test
Total iterations: 20
Passed: 18
Failed: 2
Success rate: 90.0%

Duration statistics:
  Average: 2.45s
  Min: 2.12s
  Max: 4.23s

Failed iterations: [7, 14]
```

## Testing

Tested locally with:
- ✅ Single test multiple iterations
- ✅ Known flaky tests mode
- ✅ Timeout configuration
- ✅ Failure logging

## Documentation

- Added section to main README.md
- Created comprehensive FLAKY_TEST_RUNNER.md
- Added Makefile targets with help text

## Related

Closes #138

### CODE REVIEW

**Comment by hrfarmer** [comment]
Just a heads up, I'm working on redoing some of the integration tests in #321 to hopefully fix their flakyness. Don't think these will conflict though, just wanted to make sure you're aware.

Also I'm pretty sure the linked issue is somewhat outdated to the tests on main that are flaky now

File: FprimeZephyrReference/test/int/FLAKY_TEST_RUNNER.md [hunk 1]
```diff
@@ -0,0 +1,264 @@
...
+1. **Build the firmware:**
+   ```bash
+   make build
+   ```
+
+2. **Flash firmware to board:**
+   ```bash
+   cp build-artifacts/zephyr.uf2 /Volumes/RPI-RP2
+   ```
+
```
[the remark this answers is not in the document]
Comment (yudataguy) [reply FprimeZephyrReference/test/int/FLAKY_TEST_RUNNER.md]: this was copied from main README, since that place was outdated before I found out. will resolve both issues.
